pkg/controllers/machine_controller.go

```diff
@@ -173,6 +185,33 @@ func (r *MachineReconciler) Start(ctx context.Context) error {
 	return nil
 }
 
+func (r *MachineReconciler) startCheckAndEnqueueMachineStatus(ctx context.Context, log logr.Logger) {
+	wait.UntilWithContext(ctx, func(ctx context.Context) {
+		machines, err := r.machines.List(ctx)
+		if err != nil {
+			log.Error(err, "failed to list machines")
+			return
+		}
+
+		for _, machine := range machines {
+			if machine.DeletedAt != nil || !slices.Contains(machine.Finalizers, MachineFinalizer) {
+				continue
+			}
+			id := machine.ID
+
+			state, err := r.getMachineState(id)
+			if err != nil {
+				log.Error(err, "failed to get machine state", "machine:", id)
+				continue
+			}
+
+			if state != machine.Status.State {
```

Comment:
how solve it your issue with stucked domain reconcilation loop

Reply:
In store, the machine state will be "Pending". The synchronizer will get the state as "Running" and put it to the queue. (This is where it solves the stuck domain reconciliation loop)
In the next reconciliation, the correct machine state will be set in the end.

I have tested it and working fine.